src/engines/mod.rs

```diff
@@ -703,29 +704,29 @@ enum FileFormat {
     Vf,
 }
 
-fn format_to_extension (format: FileFormat) -> &'static str {
+fn format_to_extension (format: FileFormat) -> Vec<&'static str> {
     match format {
-        FileFormat::AFM => ".afm",
-        FileFormat::Bib => ".bib",
-        FileFormat::Bst => ".bst",
-        FileFormat::Cmap => ".cmap", /* XXX: kpathsea doesn't define any suffixes for this */
-        FileFormat::Enc => ".enc",
-        FileFormat::Format => ".fmt.gz",
-        FileFormat::FontMap => ".map",
-        FileFormat::MiscFonts => ".miscfonts", /* XXX: no kpathsea suffixes */
-        FileFormat::Ofm => ".ofm", /* XXX: also .tfm */
-        FileFormat::OpenType => ".otf", /* XXX: also OTF */
-        FileFormat::Ovf => ".ovf", /* XXX: also .vf */
-        FileFormat::Pict => ".pdf", /* XXX: also .eps, .epsi, ... */
-        FileFormat::Pk => ".pk",
-        FileFormat::ProgramData => ".programdata", /* XXX no suffixes */
-        FileFormat::Sfd => ".sfd",
-        FileFormat::Tex => ".tex", /* also .{sty,cls,fd,aux,bbl,def,clo,ldf} */
-        FileFormat::TexPsHeader => ".pro",
-        FileFormat::TFM => ".tfm",
-        FileFormat::TrueType => ".ttf", /* XXX: also .ttc, .TTF, .TTC, .dfont */
-        FileFormat::Type1 => ".pfa", /* XXX: also .pfb */
-        FileFormat::Vf => ".vf",
+        FileFormat::AFM => vec!["afm"],
+        FileFormat::Bib => vec!["bib"],
+        FileFormat::Bst => vec!["bst"],
+        FileFormat::Cmap => vec!["cmap"], /* XXX: kpathsea doesn't define any suffixes for this */
+        FileFormat::Enc => vec!["enc"],
+        FileFormat::Format => vec!["fmt.gz"],
+        FileFormat::FontMap => vec!["map"],
+        FileFormat::MiscFonts => vec!["miscfonts"], /* XXX: no kpathsea suffixes */
+        FileFormat::Ofm => vec!["ofm"], /* XXX: also .tfm */
```

Comment:
@pkgw I tried adding `tfm` here too, but it turns out that adding it here fails the `test_space` test -- the code actually fails at this point (https://github.com/tectonic-typesetting/tectonic/blob/master/tectonic/dpx-tfm.c#L579).

I therefore omitted it here.